### ISSUE TITLE
Publish PPA automatically

### ISSUE THREAD

**Issue by fujiapple852** [fujiapple852/trippy#1088]
Inspired by:

https://github.com/yuezk/publish-ppa-package/blob/main/build.sh

This script needs to be updated to reflect the cargo vendoring step needed (see https://github.com/fujiapple852/trippy/blob/master/debian/README.debian).

Note that this script is able to publish for all supported Ubuntu versions.

**Comment by fujiapple852** [comment]
Step (roughyl) to publish manually (for 0.10.0 on `Jammy`):

```shell
docker run -it -v (pwd):/data ubuntu:jammy
```

```shell

apt update && apt install build-essential devscripts debhelper cargo wget

cd /data

wget https://github.com/fujiapple852/trippy/archive/refs/tags/0.10.0.tar.gz
mv 0.10.0.tar.gz.1 trippy_0.10.0.orig.tar.gz

mkdir build
mv debian build
cd build

# build vendored dep crates
./debian/rules vendor

# setup gpg and import launchpad key pair
# gpg --import ...

# build PPA source package
debuild --prepend-path ~/.cargo/bin -S -sa

# publish to launchpad
dput ppa:fujiapple/trippy ../trippy_0.10.0-1ubuntu0.1~jammy1_source.changes
```

**Comment by fujiapple852** [comment]
Issues found when publishing 0.11.0.

1: The downloaded tarball doesn't have the `.1` suffix:

```
mv 0.10.0.tar.gz.1 trippy_0.10.0.orig.tar.gz
```

I just changed the command to remove it.

2: the `changelog` file needs to be updated for each release, rather than appended to.  I did this by hand and manually patched the tar file as I had already cut the github release before updating the version.

```
trippy (0.11.0-1ubuntu0.1~jammy1) jammy; urgency=medium

  * 0.11.0 release.

 -- Fuji Apple <fujiapple852@gmail.com>  Sun, 11 Aug 2024 20:55:00 +0800
```

3: For `gpg --import` I copied the `launchpad_secret_key.pgp` to `build` and imported with `gpg --import launchpad_secret_key.pgp`

4: debuild failed with:

```
error: unwanted binary file: debian/trippy_0.11.0.orig.tar.gz
```

Because I had debian/trippy_0.11.0.orig.tar.gz as well as in the git repo root (the parent dir of `build`). We need the later, not the former, so deleted it.

5: debuild failed with: 

```
dpkg-source: error: aborting due to unexpected upstream changes, see /tmp/trippy_0.11.0-1ubuntu0.1~jammy1.diff.NQ5XoT
dpkg-source: info: Hint: make sure the version in debian/changelog matches the unpacked source tree
```

This is the point when I manually patched the tar file (#2 above), but that didn't fix it (but still needed, next time do it before the github release).  The actual error was:

```
dpkg-source: warning: file build/launchpad_secret_key.pgp has no final newline (either original or modified version)
dpkg-source: info: local changes detected, the modified files are:
 build/launchpad_secret_key.pgp
```

I moved the key to the parent directory and that fixed it.
